README.md

```diff
@@ -13,70 +13,126 @@ npm install @shopify/shopify-function-test-helpers
+```
 
-// Build a function
-const payload = buildFunction();
+### Complete Test Example
 
-// Run a function implementation
+For a full test suite that runs multiple fixtures:
 
-const result = runFunction(payload, mockFunction);
-console.log('Result status:', result.status);
+```javascript
+import path from "path";
+import fs from "fs";
+import {
+  buildFunction,
+  loadFixture,
+  runFunction,
+  validateTestAssets,
+  loadSchema,
+  loadInputQuery,
+} from "@shopify/shopify-function-test-helpers";
+
+describe("Function Tests", () => {
+  let schema;
+  let inputQueryAST;
+  let functionDir;
+
+  beforeAll(async () => {
+    functionDir = path.dirname(__dirname);
+    await buildFunction(functionDir);
+
+    const schemaPath = path.join(functionDir, "schema.graphql");
+    const inputQueryPath = path.join(functionDir, "src/run.graphql");
+
+    schema = await loadSchema(schemaPath);
+    inputQueryAST = await loadInputQuery(inputQueryPath);
+  }, 20000);
+
+  const fixturesDir = path.join(__dirname, "fixtures");
+  const fixtureFiles = fs
+    .readdirSync(fixturesDir)
+    .filter((file) => file.endsWith(".json"))
+    .map((file) => path.join(fixturesDir, file));
+
+  fixtureFiles.forEach((fixtureFile) => {
+    test(`runs ${path.basename(fixtureFile)}`, async () => {
+      const fixture = await loadFixture(fixtureFile);
+
+      const validationResult = await validateTestAssets({
+        schema,
+        fixture,
+        inputQueryAST,
+      });
+
+      expect(validationResult.inputQuery.valid).toBe(true);
+      expect(validationResult.inputFixture.valid).toBe(true);
+      expect(validationResult.inputQueryFixtureMatch.valid).toBe(true);
+      expect(validationResult.outputFixture.valid).toBe(true);
+
+      const runResult = await runFunction(
+        fixture.export,
+        fixture.input,
+        functionDir
+      );
+
+      expect(runResult.error).toBeNull();
+      expect(runResult.result.output).toEqual(fixture.expectedOutput);
+    }, 10000);
+  });
+});
 ```
 
 ## API Reference
 
-### Functions
-
-#### `loadFixture(filename)`
-Loads a fixture from the test_data directory.
-
-**Parameters:**
-- `filename`: Name of the fixture file
-
-**Returns:** Fixture object
-
-**Throws:** Error if fixture file is not found
-
-#### `validateFixture(fixture)`
-Validates a fixture to ensure it has the correct structure.
-
-**Parameters:**
-- `fixture`: The fixture data to validate
+### Core Functions
 
-**Returns:** ValidationResult object with `isValid` boolean and `errors` array
+- [`loadFixture`](./src/methods/load-fixture.ts) - Load a fixture file from the specified path
+- [`loadSchema`](./src/methods/load-schema.ts) - Load a GraphQL schema from a file
+- [`loadInputQuery`](./src/methods/load-input-query.ts) - Load and parse a GraphQL input query
+- [`validateTestAssets`](./src/methods/validate-fixture.ts) - Validate test assets including input query, fixture input/output, and query-fixture match
```

Comment:
Should this actually be called `validate-test-assets.ts`?

Reply:
Ty, nice catch!